experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -52,13 +52,37 @@ export abstract class InstrumentationBase<T = any>
         'No modules instrumentation has been defined,' +
         ' nothing will be patched'
       );
+    } else {
+      this._warnOnPreloadedModules();
     }
 
     if (this._config.enabled) {
       this.enable();
     }
   }
 
+  private _warnOnPreloadedModules(): void {
+    const preloadedModules: string[] = [];
+    this._modules.forEach((module: InstrumentationModuleDefinition<T>) => {
+      const { name } = module;
+      try {
+        const resolvedModule = require.resolve(name);
+        if (require.cache[resolvedModule]) {
+          // Module is already cached, which means the instrumentation hook might not work
+          preloadedModules.push(name);
+        }
+      } catch {
+        // Module isn't available, we can simply skip
+      }
+    });
+
+    if (!preloadedModules.length) {
+      return;
+    }
+
+    this._diag.warn(`Some modules (${preloadedModules.join(', ')}) were already required when their respective plugin was loaded, some plugins might not work. Make sure the SDK is setup before you require in other modules.`);
```

Comment:
Yeah i think its better to log once per instrumentation and to just report that the module instrumentation want to patch is already loaded

Reply:
> Yeah i think its better to log once per instrumentation and to just report that the module instrumentation want to patch is already loaded

We don't write the instrumentation name in the message itself and rely on it being in the diag scope, right? How will that look in practice?

Reply:
I suggested `Module ${name} has been loaded before ${instrumentationName} so it might not work, please try creating it before requiring ${name}` to Ran, i think thats clear enough ? (Maybe reword `creating it`, i'm not sure end user know they are creating it anyway)

Reply:
I like it 👍

Reply:
I think still stand @nozik, we need to only log the problematic module for each instrumentation instead of all of them

Reply:
> I think still stand @nozik, we need to only log the problematic module for each instrumentation instead of all of them

There is an if statement above that `return` when `!preloadedModules.length`, so I think it will only log problematic modules.

However, I still support making the change you proposed above to the message. @nozik what do you think?

Reply:
Done